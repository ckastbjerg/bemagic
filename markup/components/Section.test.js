/* eslint-env jest */
const Section = require('./Section');

test('it works for components without states', () => {
    global.config = require('../../config');

    expect(Section({
        markup: '<div class="block">The quick brown fox jumps over the lazy dog</div>',
        atRules: {},
        classes: 'block',
        mixedStates: {},
        name: 'block',
        pseudoStates: {},
        states: {},
        tagName: 'div',
    })).toMatchSnapshot();
});


test('it returns the expected markup', () => {
    global.config = require('../../config');

    expect(Section({
        atRules: {},
        classes: 'block',
        markup: '<div class="block">The quick brown fox jumps over the lazy dog</div>',
        mixedStates: {
            hoverdimmed: {
                atRules: {},
                pseudo: 'hover',
                state: 'dimmed',
            },
        },
        name: 'block',
        pseudoStates: {
            hover: 'hover',
        },
        states: {
            dimmed: {
                atRules: {},
                name: 'dimmed',
            },
        },
        tagName: 'div',
    })).toMatchSnapshot();
});
