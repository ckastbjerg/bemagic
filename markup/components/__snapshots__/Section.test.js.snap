exports[`test it returns the expected markup 1`] = `
"
        <div class=\"bemagic-page__section js-page-section background\" data-class=\"block\">
            <div class=\"block\">The quick brown fox jumps over the lazy dog</div>
            
            
                    <div    class=\"block pseudo-hover\">
                        hover
                    </div>
                
            
                    <div    class=\"block pseudo-hover is-dimmed\">
                        dimmed:hover
                    </div>
                
        </div>
    "
`;

exports[`test it works for components without states 1`] = `
"
        <div class=\"bemagic-page__section js-page-section background\" data-class=\"block\">
            <div class=\"block\">The quick brown fox jumps over the lazy dog</div>
            
            
            
        </div>
    "
`;
